README.md
![Tabetalt Kit][banner]

![Build & Deploy][build-badge]
[![Total alerts][lgtm-badge]][lgtm-alerts]
[![LGTM Grade][lgtm-grade]][lgtm-alerts]
[![Maintainability][codeclimate-badge]][codeclimate]
[![npm bundle size][bundle-size]][npm]

## About

Tabetalt Kit is the technical implementation of Tabetalt Designsystem. It is based on [Theme UI][theme-ui] and features theme and a collection of components for use in Tabetalt applications.

Our intentions are to stylize according to _Tabetalt Designsystem_ on all [Theme UI components][theme-ui-components].

## Quickstart

All components are included in one package. Install it as such:

```shell
▶ yarn add @tabetalt/kit
```

Install necessary peer dependencies.

```shell
▶ yarn add theme-ui@next
```

**Remember to include Google Fonts** No fonts are packaged with the library. Therefore, you'll need to add a Link to it, as such:

```html
<link
  href="https://fonts.googleapis.com/css2?family=DM+Sans:ital,wght@0,400;0,700;1,400;1,700&display=swap"
  rel="stylesheet"
/>
```

## Example

You'll have to register the `ThemeProvider` in order to make the theme work.

```jsx
import { ThemeProvider, Styled } from 'theme-ui';
import { theme } from '@tabetalt/kit';

const App = () => (
  <ThemeProvider theme={theme}>
    <Styled.h1>Hello world!</Styled.h1>
  </ThemeProvider>
);
```

## Documentation

Apart from this README, you can find details and examples of using the SDK in
the following places:

- [Storybook][storybook]
- [Theme UI][theme-ui]
- [Theme UI Components][theme-ui-components]

[banner]: ./assets/banner.jpg
[npm]: https://www.npmjs.com/package/@tabetalt/kit
[build-badge]: https://img.shields.io/github/workflow/status/tabetalt/kit/Build%20and%20Deploy
[storybook]: https://kit.tabetalt.no/
[storybook-badge]: https://raw.githubusercontent.com/storybookjs/brand/master/badge/badge-storybook.svg
[theme-ui]: https://github.com/system-ui/theme-ui
[theme-ui-components]: https://theme-ui.com/components
[codeclimate-badge]: https://img.shields.io/codeclimate/maintainability/tabetalt/kit
[codeclimate]: https://codeclimate.com/github/tabetalt/kit/maintainability
[lgtm-badge]: https://img.shields.io/lgtm/alerts/g/tabetalt/kit.svg?logo=lgtm&logoWidth=18
[lgtm-alerts]: https://lgtm.com/projects/g/tabetalt/kit/alerts/
[lgtm-grade]: https://img.shields.io/lgtm/grade/javascript/github/tabetalt/kit
[bundle-size]: https://img.shields.io/bundlephobia/min/@tabetalt/kit


CHANGELOG.md
## [1.21.1](https://github.com/tabetalt/kit/compare/v1.21.0...v1.21.1) (2021-08-18)


### Bug Fixes

* add `dist` to `"files"` ([ac63818](https://github.com/tabetalt/kit/commit/ac638186cd4c82cd8637cee4e9da9c83d0747e60))

# [1.21.0](https://github.com/tabetalt/kit/compare/v1.20.0...v1.21.0) (2021-08-18)


### Features

* Add ESM/CJS ([90c3f41](https://github.com/tabetalt/kit/commit/90c3f419531b04e63ce0284328ff7bd7121b60a3))

# [1.20.0](https://github.com/tabetalt/kit/compare/v1.19.9...v1.20.0) (2021-05-09)


### Bug Fixes

* Fix Logout size. ([ec492f1](https://github.com/tabetalt/kit/commit/ec492f11c328709b6c5e60b99d2697a912fb3a67))


### Features

* Add logout icon. ([2bd7851](https://github.com/tabetalt/kit/commit/2bd7851b57939245789bcee305f8e6d7b4562f94))

## [1.19.9](https://github.com/tabetalt/kit/compare/v1.19.8...v1.19.9) (2020-11-19)


### Bug Fixes

* **FileInput:** FileInput component wrapper width fix ([#92](https://github.com/tabetalt/kit/issues/92)) ([feb2439](https://github.com/tabetalt/kit/commit/feb2439cef39d77a4a29afdc488a9937e1739d29))

## [1.19.8](https://github.com/tabetalt/kit/compare/v1.19.7...v1.19.8) (2020-11-18)


### Bug Fixes

* **input-tag:** fixed z-index of suggetions ([c683c8c](https://github.com/tabetalt/kit/commit/c683c8c9cb6d997563259819c4e3099a98c33861))

## [1.19.7](https://github.com/tabetalt/kit/compare/v1.19.6...v1.19.7) (2020-11-18)


### Bug Fixes

* **editor-inputTag:** added default value for editor. ([6f99ed9](https://github.com/tabetalt/kit/commit/6f99ed9aea284a960b33b5b0e96ab5b4a853e12f))

## [1.19.6](https://github.com/tabetalt/kit/compare/v1.19.5...v1.19.6) (2020-11-16)


### Bug Fixes

* **editor:** fixed default value component ([170fc4c](https://github.com/tabetalt/kit/commit/170fc4cc525c83fb301631111ad5ca7379cc1bd1))
* **editor-serialize-text:** added parsing html text for editor ([7534bc7](https://github.com/tabetalt/kit/commit/7534bc74a92bc46260cb30613d52dc62a4b3b18d))

## [1.19.5](https://github.com/tabetalt/kit/compare/v1.19.4...v1.19.5) (2020-11-11)


### Bug Fixes

* **PrefilledInput:** Fixed PrefilledInput change, focus, blur events handling ([1c40c59](https://github.com/tabetalt/kit/commit/1c40c59cc435f360f87547ecb33d98fe5dc40422))

## [1.19.4](https://github.com/tabetalt/kit/compare/v1.19.3...v1.19.4) (2020-11-11)


### Bug Fixes

* **prefilled-input:** added firing on change event ([a8dfad9](https://github.com/tabetalt/kit/commit/a8dfad99188f6da9e5b1f474fc36193d28d70d96))
* **prefilled-input:** changed ref to prefilled input ([95ebbf9](https://github.com/tabetalt/kit/commit/95ebbf9f9b0e24542b3e1dd4e31e36d4f14ed357))

## [1.19.3](https://github.com/tabetalt/kit/compare/v1.19.2...v1.19.3) (2020-11-10)


### Bug Fixes

* **editor:** fixed editor border color ([0a9b842](https://github.com/tabetalt/kit/commit/0a9b842d8180cecfbea0e6bda9df676054714f84))

## [1.19.2](https://github.com/tabetalt/kit/compare/v1.19.1...v1.19.2) (2020-11-08)


### Bug Fixes

* **switch:** fixed switch colors ([1d26ccc](https://github.com/tabetalt/kit/commit/1d26ccc16e147435e93ebb9a447a54e310f846f9))

## [1.19.1](https://github.com/tabetalt/kit/compare/v1.19.0...v1.19.1) (2020-11-06)


### Bug Fixes

* **Button:** Improved Button, IconButton styles ([a64087e](https://github.com/tabetalt/kit/commit/a64087e366c2688921258d03fb1c9f91f773eba9))

# [1.19.0](https://github.com/tabetalt/kit/compare/v1.18.0...v1.19.0) (2020-11-06)


### Bug Fixes

* **Field:** changed props  passing ([9b60f9a](https://github.com/tabetalt/kit/commit/9b60f9a8027fb3f58b7926f856586718d7245e2f))
* **Field:** fixed filed alignment ([95111a1](https://github.com/tabetalt/kit/commit/95111a12cbe19cfb5e63b73fc8ba3a27f9999c60))
* **FileUpload:** fixed export from FileUpload component ([da790f0](https://github.com/tabetalt/kit/commit/da790f0373c9b8817b630409766d43dd65faec7f))
* **Theme:** changes bg color ([8a35781](https://github.com/tabetalt/kit/commit/8a35781226390eeca281738529770b35edf66c57))


### Features

* **StatusLabel:** added status label component ([94575b2](https://github.com/tabetalt/kit/commit/94575b2c53033323f402546800a09a65db352cbb))

# [1.18.0](https://github.com/tabetalt/kit/compare/v1.17.0...v1.18.0) (2020-10-29)


### Bug Fixes

* **FileUpload:** removed unnecessary button file ([c609948](https://github.com/tabetalt/kit/commit/c6099487227d03af56a5ccfa8cd02a0c410fb627))
* **LabeledSelect:** fix export with labeled select ([598912b](https://github.com/tabetalt/kit/commit/598912b53b7a2cb6dc294be05320c4d52302d44a))


### Features

* **FileUpload:** Added basic FileUpload compinent ([2a96f56](https://github.com/tabetalt/kit/commit/2a96f566eb65e61d3b7988d0f22fb84968c1743d))
* **FileUpload:** Improved FileUpload component, added strories ([463aa8a](https://github.com/tabetalt/kit/commit/463aa8a94394d8147aedd22ba7196db63c8f0857))

# [1.17.0](https://github.com/tabetalt/kit/compare/v1.16.0...v1.17.0) (2020-10-29)


### Bug Fixes

* **Editor:** fixed editor border style ([357d1d3](https://github.com/tabetalt/kit/commit/357d1d3a86953d328d0a26fb13b57fa179036efe))
* **Field:** fixed field alignment ([92b915b](https://github.com/tabetalt/kit/commit/92b915bb4486d23e333bff8c6a1f6bfaa1a3f7ab))
* **PrefilledInput:** fixed focus to element ([4707f44](https://github.com/tabetalt/kit/commit/4707f44f9c1815713d23db1a66a2e5fd369b5cab))


### Features

* **Editor:** added logic for hiding toolbar ([76681f8](https://github.com/tabetalt/kit/commit/76681f89246a5d8f69d7b48b319990587afd97eb))

# [1.16.0](https://github.com/tabetalt/kit/compare/v1.15.0...v1.16.0) (2020-10-28)


### Features

* **LabeledSelect:** added select comp with label ([1534111](https://github.com/tabetalt/kit/commit/15341119a8fa1903f0f24cb8ea61a4130aa65af1))

# [1.15.0](https://github.com/tabetalt/kit/compare/v1.14.2...v1.15.0) (2020-10-27)


### Bug Fixes

* **PrefilledInput:** fixed export statement ([37a7b0c](https://github.com/tabetalt/kit/commit/37a7b0cba0c7134d5573732aa127516baa1f9343))


### Features


src/theme.ts
/* eslint-disable @typescript-eslint/no-non-null-assertion */
import { Theme } from 'theme-ui';

const defaultBadge = {
  backgroundColor: 'gray_1',
  borderRadius: '38px',
  color: 'text',
  py: 1,
  px: 3,
};

const badges = {
  primary: defaultBadge,
  success: {
    ...defaultBadge,
    backgroundColor: 'success',
    color: 'background',
  },
  lightest: {
    ...defaultBadge,
    backgroundColor: 'gray_2',
    color: 'gray_0',
  },
  warning: {
    ...defaultBadge,
    backgroundColor: 'primary',
    color: 'background',
  },
  darker: {
    ...defaultBadge,
    backgroundColor: 'secondary',
    color: 'background',
  },
  error: {
    ...defaultBadge,
    backgroundColor: 'error',
    color: 'background',
  },
};

const grayscale = ['#9AA0B5', '#E4E6EC', '#EEEFF3', '#F6F6F9'];

const theme = {
  breakpoints: ['40em', '52em', '64em'],
  space: [0, 4, 8, 16, 32, 64, 128, 256, 512],
  fonts: {
    body:
      '"DM Sans", system-ui, -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, "Helvetica Neue", sans-serif',
    heading: 'inherit',
    monospace: 'Menlo, monospace',
  },
  fontSizes: [12, 14, 16, 20, 24, 32, 48, 64],
  fontWeights: {
    body: 400,
    heading: 700,
    bold: 700,
  },
  lineHeights: {
    body: 1.5,
    heading: 1.125,
  },
  buttons: {
    primary: {
      py: 2,
      px: 3,
      fontFamily: 'body',
      borderRadius: '38px',
      outline: 'none',
      cursor: 'pointer',
      transition: 'background-color 0.3s',
      '&:active': {
        bg: '#E39256',
      },
      '&:disabled,&[disabled]': {
        cursor: 'default',
        bg: 'highlight',
        color: 'muted',
      },
    },
    outline: {
      py: '8px',
      px: 3,
      fontFamily: 'body',
      borderRadius: '38px',
      outline: 'none',
      cursor: 'pointer',
      transition: 'background-color 0.3s, box-shadow 0.3s',
      bg: 'transparent',
      boxShadow: 'inset 0 0 0 3px var(--theme-ui-colors-secondary, #1B202E)',
      // border: '3px solid',
      // borderColor: 'secondary',
      color: 'text',
      '&:active': {
        bg: 'gray_2',
      },
      '&:hover': {
        boxShadow: 'inset 0 0 0 3px var(--theme-ui-colors-secondary, #FFA665)',
      },
      '&:disabled,&[disabled]': {
        cursor: 'default',
        bg: 'highlight',
        boxShadow: 'inset 0 0 0 3px var(--theme-ui-colors-muted, #9AA0B5)',
        color: 'muted',
      },
    },
    warning: {
      py: '8px',
      px: 3,
      fontFamily: 'body',
      borderRadius: '38px',
      outline: 'none',
      cursor: 'pointer',
      bg: 'transparent',
      transition: 'background-color 0.3s, box-shadow 0.3s',
      boxShadow: 'inset 0 0 0 3px var(--theme-ui-colors-errors, #E03B54)',
      // border: '3px solid',
      // borderColor: 'secondary',
      color: 'error',
      '&:active': {
        bg: 'gray_2',
      },
      '&:disabled,&[disabled]': {
        cursor: 'default',
        bg: 'highlight',
        boxShadow: 'inset 0 0 0 3px var(--theme-ui-colors-error, #E03B54)',
        color: 'muted',
      },
    },
    destructive: {
      py: '8px',
      px: 3,
      fontFamily: 'body',
      borderRadius: '38px',
      outline: 'none',
      cursor: 'pointer',
      bg: 'error',
      color: 'background',
      transition: 'background-color 0.3s',
      '&:active': {
        bg: '#CB374D',
      },
      '&:disabled,&[disabled]': {
        cursor: 'default',
        bg: 'highlight',
        color: 'muted',
      },
    },
  },
  colors: {
    text: '#1B202E',
    background: '#F6F6F9',
    primary: '#FFA665',
    secondary: '#1B202E',
    accent: '#2B40B5',
    highlight: '#E4E6EC',
    muted: '#9AA0B5',
    ...grayscale.reduce((colors: any, curColor, i) => {
      colors[`gray_${i}`] = curColor;
      return colors;
    }, {}),
    error: '#E03B54',
    success: '#00CD98',
  },
  text: {
    heading: {
      fontFamily: 'heading',
      lineHeight: 'heading',
      fontWeight: 'heading',
    },
  },
  links: {
    focus: {
      textDecoration: 'none',
      color: 'secondary',
      border: '1px solid',
      borderColor: 'secondary',
      p: 1,
    },
    visited: {
      textDecoration: 'underline',
      color: 'muted',
    },
    hover: {
      textDecoration: 'underline',
      color: 'primary',
      cursor: 'pointer',
    },
  },
  forms: {
    checkbox: {
      'input:checked ~ &': { color: 'secondary' },
      'input:disabled ~ &': {
        color: 'gray_1',
        bg: 'gray_2',
      },
    },
    radio: {
      'input:checked ~ &': { color: 'secondary' },
      'input:disabled ~ &': {
        color: 'gray_1',
        bg: 'gray_2',
      },
    },
    input: {
      fontFamily: 'body',
      borderRadius: '38px',
      border: '1px solid',
      borderColor: 'gray_1',
      px: 4,
      py: 3,
      '&:focus,&:not(:placeholder-shown)': {
        outline: 'none',
        borderColor: 'gray_0',
      },
      '::placeholder,:-ms-input-placeholder,::-ms-input-placeholder': {
        color: 'gray_0',
      },
      '&:disabled,&[disabled]': {
        bg: 'highlight',
        color: 'muted',
      },
    },
  },
  badges,
  styles: {
    root: {
      fontFamily: 'body',
      lineHeight: 'body',
      fontWeight: 'body',
      color: 'text',
      '*': { boxSizing: 'border-box' },
    },
    h1: {
      variant: 'text.heading',
      fontSize: 7,
    },
    h2: {
      variant: 'text.heading',
      fontSize: 6,
    },
    h3: {
      variant: 'text.heading',
      fontSize: 5,
    },
    h4: {
      variant: 'text.heading',
      fontSize: 4,
    },
    h5: {
      variant: 'text.heading',
      fontWeight: 'normal',
      fontSize: 4,
    },
    h6: {
      variant: 'text.heading',
      textTransform: 'uppercase',
      fontWeight: 'normal',
      fontSize: 2,
    },
    pre: {
      fontFamily: 'monospace',
      overflowX: 'auto',
      code: {
        color: 'inherit',
      },
    },
    code: {
      fontFamily: 'monospace',
      fontSize: 'inherit',
    },
    table: {
      width: '100%',
      borderCollapse: 'separate',
      borderSpacing: 0,
    },
    th: {
      textAlign: 'left',
      borderBottomStyle: 'solid',
    },
    td: {
      textAlign: 'left',
      borderBottomStyle: 'solid',
    },
    a: {
      color: 'accent',
      textDecoration: 'underline',
      '&:hover': {
        variant: 'links.hover',
      },
      '&:focus': {
        variant: 'links.focus',
      },
      '&:visited': {
        variant: 'links.visited',
      },
    },
  },
} as Theme;

theme.forms!.error = {
  ...theme.forms!.input,
  borderColor: 'error',
  '&:focus,&:not(:placeholder-shown)': {
    outline: 'none',
    borderColor: 'error',
  },
};

export default theme;
